common/placer1.cc

```diff
@@ -51,15 +51,20 @@ class SAPlacer
     {
         int num_bel_types = 0;
         for (auto bel : ctx->getBels()) {
-            Loc loc = ctx->getBelLocation(bel);
             IdString type = ctx->getBelType(bel);
-            int type_idx;
             if (bel_types.find(type) == bel_types.end()) {
-                type_idx = num_bel_types++;
-                bel_types[type] = type_idx;
+                bel_types[type] = std::tuple<int, int>(num_bel_types++, 1);
             } else {
-                type_idx = bel_types.at(type);
+                std::get<1>(bel_types.at(type))++;
             }
+        }
+        for (auto bel : ctx->getBels()) {
+            Loc loc = ctx->getBelLocation(bel);
+            IdString type = ctx->getBelType(bel);
+            int type_idx = std::get<0>(bel_types.at(type));
+            int type_cnt = std::get<1>(bel_types.at(type));
+            if (type_cnt < 64)
```

Comment:
Can you make this magic number a constant somewhere? This might even be something that the arch API should supply. 64 seems a bit large to me, as this will also include block RAM - and DSPs on the ECP5.

Reply:
Fixed with `Placer1Cfg.minBelsForGridPick`